web/docs/internal/keystroke-lifecycle.md

```diff
@@ -0,0 +1,90 @@
+# The Keyman Engine for Web keystroke lifecycle
+
+Depending on the variant of Keyman Engine for Web in play, the engine integrates either with a host app or with the current browser page to interpret and/or intercept keystroke events and produce the desired context manipulations corresponding to them.  While some details diverge between `app/browser` and `app/webview` builds, the general structure and pattern for this process holds.
+
+This document will break down the full keystroke lifecycle into four distinct phases, as follows:
+
+1. Keystroke pre-processing: event interception + interpretation (pre-processing)
+2. Keystroke processing: keystroke main-group rule evaluation
+3. Keystroke post-processing: keystroke post-keystroke evaluation + side-effects
+4. Keystroke resolution: UI state updating
+
+## Keystroke pre-processing
+
+Both the `app/browser` and `app/webview` products are capable of handling simulated keystrokes from the on-screen keyboard as well as hardware keystrokes.  For hardware keystrokes, this varies between the two, while the majority of on-screen keyboard keystrokes are handled identically.
+
+### `app/browser`
+
+The `app/browser` version of the engine is designed to integrate with the active page of the user's browser, giving it far more direct access to the user's true context in addition to direct access to handling DOM events resulting from hardware keystrokes.  Its `HardwareEventKeyboard` class is the primary module for establishing the needed engine event hooks and for interpreting the corresponding DOM events as they arise.
```

Comment:
I don't really understand what 'far more direct access to the user's true context' means? Far more direct than what? How does this change the way the keystroke lifecycle works?